### PR TITLE
Handle short form queries

### LINKED ISSUE #2855: Support deserialization of query short form
Relates #2836 

Deserialization in NEST should handle the short form of queries such that it is able to correctly deserialize queries that have been constructed outside of NEST.

### PR DESCRIPTION
**Pull request by russcam**
This commit handles the short form of the following queries:

- term query
- match query
- prefix query
- match phrase query
- match phrase prefix query
- match bool prefix query

Closes #2855

### CODE REVIEW

File: src/Nest/QueryDsl/Abstractions/FieldName/FieldNameQueryFormatter.cs [hunk 1]
```diff
@@ -34,18 +36,66 @@ public override void Serialize(ref JsonWriter writer, TInterface value, IJsonFor
...
+										termQuery.Value = NumberConverter.ReadDouble(segment.Array, segment.Offset, out var count);
+									break;
+								case JsonToken.True:
+								case JsonToken.False:
+									termQuery.Value = reader.ReadBoolean();
+									break;
+							}
+							reader.ReadIsEndObjectWithVerify();
+							break;
+						case IMatchQuery matchQuery:
```
Comment (Mpdreamz) [inline src/Nest/QueryDsl/Abstractions/FieldName/FieldNameQueryFormatter.cs]: We need more cases while we are it:

https://github.com/elastic/clients-flight-recorder/blob/master/script/lib/normalizer.js#L430-L434

File: tests/Tests/QueryDsl/FullText/Match/MatchUsageTests.cs [hunk 2]
```diff
@@ -69,5 +74,16 @@ public class MatchUsageTests : QueryDslUsageTestsBase
 				.Name("named_query")
 				.AutoGenerateSynonymsPhraseQuery(false)
 			);
+
+		//hide
+		[U] public void DeserializeShortForm()
+		{
+			var json = JsonConvert.SerializeObject(new { description = "project description" });
```
Comment (Mpdreamz) [inline tests/Tests/QueryDsl/FullText/Match/MatchUsageTests.cs]: Move to base and reuse for other DeserializeShortForm tests

File: src/Nest/QueryDsl/Abstractions/FieldName/FieldNameQueryFormatter.cs (hunk 1 above)
Comment (russcam) [reply src/Nest/QueryDsl/Abstractions/FieldName/FieldNameQueryFormatter.cs]:
```json
  const fieldQueries = [
    'range', 'common', 'intervals', 'match', 'match_bool_prefix', 'match_phrase', 'match_phrase_prefix',
    'geo_bounding_box', 'geo_distance', 'geo_polygon', 'geo_shape', 'shape', 'distance_feature', 'rank_feature',
    'term', 'terms'
  ]
```

I don't think the other cases are needed. It looks like 'range', 'common', 'intervals', 'geo_bounding_box', 'geo_distance', 'geo_polygon', 'geo_shape', 'shape', 'distance_feature', 'rank_feature' and 'terms' query do not support a query short form. The `switch` statement on type of query only applies when the value assigned to the field name key is not a JSON object.

File: tests/Tests/QueryDsl/FullText/Match/MatchUsageTests.cs (hunk 2 above)
Comment (russcam) [reply tests/Tests/QueryDsl/FullText/Match/MatchUsageTests.cs]: Addressed in 5445bc9dc1ba2bbc093f43e3c982ebaaecf866ac

**Comment by russcam** [comment]
Would you mind taking another look please, @Mpdreamz ?

**Comment by russcam** [comment]
Spoke with @Mpdreamz about this offline, merging in

**Comment by russcam** [comment]
Failing tests don't look related to this change, and #4280 is open to investigate

**Comment by russcam** [comment]
ported to master in 5d7df16fa2bf22214ec331dbcf559e8d53dc8060

**Pull request merged**